lib/Exporter/Writer/KnpSnappyWriter.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Writer;
+
+use Knp\Snappy\Pdf;
+
+/**
+ * Export data in PDF format.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class KnpSnappyWriter implements WriterInterface
+{
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var string
+     */
+    private $wktmltopdfBinary;
+
+    /**
+     * @var bool
+     */
+    private $showHeaders;
+
+    /**
+     * @var bool
+     */
+    private $showBorders;
+
+    /**
+     * @var array
+     *
+     * @see http://wkhtmltopdf.org/usage/wkhtmltopdf.txt
+     */
+    private $snappyOptions;
+
+    /**
+     * @var resource
+     */
+    private $file;
+
+    /**
+     * @var string
+     */
+    private $html = '
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <meta name=ProgId content=Pdf.Document>
+    <meta name=Generator content="https://github.com/sonata-project/exporter">
+    <style></style>
+  </head>
+  <body>
+    [header-html]
+    <table>[content]</table>
+    [footer-html]
+  </body>
+</html>';
+
+    /**
+     * @throws \RuntimeException
+     *
+     * @param string $filename
+     * @param bool   $showHeaders
+     * @param bool   $showBorders
+     * @param array  $snappyOptions     KnpSnappy options (wkhtmltopdf)
+     * @param string $wkhtmltopdfBinary Path of the wkhtmltopdf binary
+     */
+    public function __construct(
+        $filename,
+        $wkhtmltopdfBinary,
+        $showHeaders = true,
+        $showBorders = true,
+        array $snappyOptions = array()
+    ) {
+        $this->filename = $filename;
+        $this->showHeaders = $showHeaders;
+        $this->showBorders = $showBorders;
+        $this->snappyOptions = $snappyOptions;
+        $this->wkhtmltopdfBinary = $wkhtmltopdfBinary;
+
+        if (is_file($filename)) {
+            throw new \RuntimeException(sprintf('The file "%s" already exists', $filename));
+        }
+
+        if (!is_executable($wkhtmltopdfBinary)) {
+            throw new \RuntimeException('The file "%s" doesn\'t exist or is not executable', $wkhtmltopdfBinary);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function open()
+    {
+        $this->file = fopen($this->filename, 'w', false);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function write(array $data)
+    {
+        $tmpFile = tmpfile();
+
+        if (true === $this->showBorders) {
```

Comment:
@Soullivaneuh : about strict checks, here is an example. Here, if `$this->showBorders === 1` instead of `true`, no borders are added. Silently. If `$this->showBorders === 0` instead of `false`, the code works, which means that using this strict checking does not even ensure the method cannot be misused with something else than a boolean

@Soullivaneuh @chalasr IMO strict checking is pointless, `if ($this->showBorders)` would be way more natural to read. If you want to do strict checking, then do it before, like this : 

```
if (!is_bool($this->showBorders)) {
// … throw exception or deprecation warning saying that we will use type hinting when dropping php < 7
}
```